src/ErrorHandler/PsrErrorHandler.php

```diff
@@ -0,0 +1,46 @@
+<?php declare(strict_types = 1);
+
+namespace Apitte\Core\ErrorHandler;
+
+use Apitte\Core\Exception\Runtime\SnapshotException;
+use Apitte\Core\Utils\Helpers;
+use Psr\Log\LoggerInterface;
+use Throwable;
+
+class PsrErrorHandler implements IErrorHandler
```

Comment:
Po API chces, aby se ti vratil vzdy JSON. Logovani je sice fajn, ale spis jako opt-in.

Tzn, ze bych normalne udelal JsonErrorHandler, kterej primarne prevede exception na JSON response. A optionally zaloguje pres logger hlasku, ale nepovinne. Tzn, logger pres setter.

Reply:
Vždy ne. Tady bude zapotřebí transformovat právě jen v případě, kdy se o vytvoření response nepostará nějaký exception decorator, například skrze transformery v negotiations.

Možná bychom měli do transformerů posílat errory čistě jako value object, ať se nestarají o rozlišování, zda jde o Client/Server/jinou exception a drží se jednotná struktura. Transformery by transformovaly jen datový formát, nikoli exception.

Implementací je víc kvůli tracy, která umí logovat přímo exceptions, a tak vytvoří i bluescreen. Psr logger dovoluje jen string. Možná by bylo fajn pro vytvoření bluescreenů mít samostatnou službu, hodila by se i v contributte/monolog a contributte/logging.

Reply:
Asi to můžu spojit do jednoho handleru. Pro bluescreeny se vytvoří služba v tracy a pro monolog se vytvoří processor. Zůstane tak jen volitelný psr logger.